src/sqlalchemy_cratedb/dialect.py

```diff
@@ -228,8 +228,12 @@ def connect(self, host=None, port=None, *args, **kwargs):
         servers = to_list(server)
         if servers:
             use_ssl = asbool(kwargs.pop("ssl", False))
-            if use_ssl:
+            # TODO: Switch to the canonical default `sslmode=prefer` later.
+            sslmode = kwargs.pop("sslmode", "disable")
+            if use_ssl or sslmode in ["allow", "prefer", "require", "verify-ca", "verify-full"]:
                 servers = ["https://" + server for server in servers]
+            if sslmode == "require":
+                kwargs["verify_ssl_cert"] = False
```

Comment:
This looks odd as it disables cert verification even that the user did not intend to do that.
If one need to do that, one can set the arg explicitly, or what do I miss?

Reply:
Currently, `verify_ssl_cert` can not be defined by the user on behalf of the SQLAlchemy connection string. Its the intention of the patch to unlock that.

Reply:
Ok, but this will disable cert verification for everyone who uses `require`. This sounds not like it is what we really want.

Reply:
We think [`sslmode=require`](https://www.postgresql.org/docs/current/libpq-ssl.html#LIBPQ-SSL-PROTECTION) is the right option to turn off host name validation on SSL connections.

The PostgreSQL docs uses uncommon jargon there, that's why it is sometimes difficult to understand, but of course we may also be wrong on this detail. [^1]

Please validate and clarify.

[^1]: We will try to use a better jargon on our docs, but still adhere to the option values PostgreSQL clients are using here.

Reply:
Ah ok, I understood it wrongly, thanks for the clarification.

Reply:
Why would we want to disable hostname validation with `sslmode=require` ?

Reply:
Hi. It's for the same reasons PostgreSQL and others offer the same option in one way or another. The specific need is originating when connecting to CrateDB from internal spots within an K8s cluster. @WalBeh can possibly explain it better.